tef-impl/src/test/java/flipkart/tef/execution/FluentCapabilityBuilderTest.java

```diff
@@ -163,6 +164,62 @@ public List<Class<? extends IBizlogic>> exclusions() {
 
     }
 
+    @Test
+    public void testThatExclusionIsHonoredWhenControlDependencyIsExcluded() throws Exception {
+
+        CapabilityDefinition capabilityDefinition = new EmptyCapabilityDefinition() {
+            @Override
+            public String name() {
+                return "C1";
+            }
+
+            @Override
+            public List<? extends CapabilityDefinition> dependentCapabilities() {
+                return Collections.emptyList();
+            }
+
+            @Override
+            public List<Class<? extends BasicValidationBizlogic>> validators() {
+                return ImmutableList.of(BasicValidationBizlogic1.class);
+            }
+
+            @Override
+            public List<Class<? extends BasicEnrichmentBizlogic>> enrichers() {
+                return ImmutableList.of(BasicEnrichmentBizlogic1.class);
+            }
+
+            @Override
+            public List<Class<? extends DataAdapterBizlogic>> adapters() {
+                return ImmutableList.of(DataAdapterBizlogic1.class);
+            }
+
+            @Override
+            public List<Class<? extends IBizlogic>> exclusions() {
+                return ImmutableList.of(DataAdapterBizlogic2.class);
+            }
+
+            @Override
+            public List<BizlogicDependency> bizlogicDependencies() {
+                List<BizlogicDependency> list = new ArrayList<>();
+                // Adding adapter 2 on adapter 1
+                list.add(new BizlogicDependency(DataAdapterBizlogic2.class, new Class[]{DataAdapterBizlogic1.class, DataAdapterBizlogic3.class}));
+                return list;
+            }
+        };
+
+        FluentCapabilityBuilder manager = new FluentCapabilityBuilder();
+        SimpleFlow flow = manager.withCapability(capabilityDefinition).dataflow();
+
+        assertNotNull(flow);
+
+        assertEquals(4, flow.getBizlogics().size());
```

Comment:
Can you also assert on the order of bizlogics?

Reply:
done